storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -245,14 +247,14 @@ private void updateChainHead(
         LOG.info("Skipping head block update to avoid potential rollback of the chain head.");
         return;
       }
-      if (originalHead.isPresent() && originalHead.get().equals(newChainHead)) {
+      if (originalHead.isPresent() && isNewHeadSameAsOld(originalHead.get(), newChainHead)) {
         LOG.trace("Skipping head update because new head is same as previous head");
         return;
       }
       this.chainHead = Optional.of(newChainHead);
       final Optional<ReorgContext> optionalReorgContext;
       if (originalHead
-          .map(head -> hasReorgedFrom(head.getRoot(), head.getForkChoiceSlot()))
+          .map(head -> hasReorgedFrom(head.getRoot(), getChainHeadSlot(head)))
```

Comment:
Probably worth updating the variable name `originalForkChoiceSlot` and comments in `hasReorgedFrom` since we're no longer always using the fork-choice slot.

Reply:
Yep, good spot.